mlflow/transformers/__init__.py

```diff
@@ -1143,6 +1149,62 @@ def _write_card_data(card_data, path):
             )
 
 
+def _list_repository_contents(model_name):
+    """Returns the top-level file inventory of `RepoFile` objects from the huggingface hub"""
+    try:
+        import huggingface_hub as hub
+    except ImportError:
+        _logger.debug(
+            f"Unable to list repository contents for the model repo {model_name}. In order "
+            "to enable repository listing functionality, please install the huggingface_hub "
+            "package by running `pip install huggingface_hub>0.10.0"
+        )
+        return
+
+    try:
+        files = hub.list_repo_tree(model_name)
+        return list(files)
+    except Exception as e:
+        _logger.debug(
+            f"Failed to retrieve repository file listing data for {model_name} due to {e}"
+        )
+
+
+def _write_license_information(model_name, card_data, path):
+    """Writes the license file or instructions to retrieve license information"""
+
+    fallback = (
+        f"A license file could not be found for the '{model_name}' repository. \n"
+        "To ensure that you are in compliance with the license requirements for this "
+        f"model, please visit the model repository here:\n https://huggingface.co/{model_name}"
+    )
+
+    if repository_contents := _list_repository_contents(model_name):
+        import huggingface_hub as hub
+
+        if license_files := [
+            file.path for file in repository_contents if file.path in _LICENSE_FILE_NAMES
+        ]:
+            try:
+                local_license_path = pathlib.Path(
+                    hub.hf_hub_download(repo_id=model_name, filename=license_files[0])
+                )
+                target_path = path.joinpath(local_license_path.name)
+                shutil.copy(local_license_path, target_path)
+                return
+            except Exception as e:
```

Comment:
I feel like we could get rid of this try catch and line 1167 by wrapping line 547 directly

Reply:
Let's minimize the scope of try-catch. `shutil.copy` failing doesn't mean `The license file could not be retrieved`.